components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java

```diff
@@ -234,4 +236,23 @@ public void updateEndpointRegistry(String registryId, String registryName, Endpo
 
     }
 
+    /**
+     * Creates a new version of an Endpoint Registry Entry
+     *
+     * @param entryId       Registry Entry Identifier(UUID)
+     * @param registryEntry EndpointRegistryEntry
+     * @param version       New version of the Registry Entry
+     * @return entryID UUID of the created Registry Entry
+     * @throws APIManagementException if failed to delete the Endpoint Registry Entry
+     */
+    public String createNewEntryVersion(String entryId, EndpointRegistryEntry registryEntry, String version)
+            throws APIManagementException {
+        registryEntry.setVersion(version);
+        if (apiMgtDAO.isRegistryEntryNameExists(registryEntry)) {
```

Comment:
When creating new version, don't we need to check whether the {name, version} combination already exists.

Reply:
**isRegistryEntryNameExists** method is now altered to do that. Shall we rename this method then to **isRegistryEntryNameVersionExists** ?

Reply:
Ah right that does not work. we need two methods. Otherwise **addRegistryEntry** resource can be used to create versions too. Will fix that

Reply:
Fixed